### PR TITLE
[GUI] Fix proposal name and URL size limit validation

### LINKED ISSUE #2666: [Bug] Wallet crashes if Amount entered in the field starts with 0
Tried on testnet with compiled wallet version of master + PR 2659, 2662, 2664 and 2665.

OS: Ubuntu 20.04

While creating a proposal using the new Governance GUI, by entering `099` in the Amount field, wallet will accept it and proceed to the next step and recognize it as 99 PIV proposal, but then the wallet crashes and the following output appears in the terminal:

```
EXCEPTION: NSt8ios_base7failureB5cxx11E
String length limit exceeded: iostream error
pivx in Runaway exception

terminate called after throwing an instance of 'std::ios_base::failure[abi:cxx11]'
  what():  String length limit exceeded: iostream error
```
  

 After trying to restart the wallet, Runaway exception appears again and wallet cannot be started, while another output shows in the terminal:
 
```
EXCEPTION: 19QUnhandledException
std::exception
pivx in Runaway exception

pivx-qt: /usr/include/boost/signals2/detail/lwm_pthreads.hpp:60: void boost::signals2::mutex::lock(): Assertion `pthread_mutexlock(&m) == 0' failed.
```

### PR DESCRIPTION
**Pull request by furszy**
Fixes #2666 crash. Which is related to the size, in bytes, of the proposal name and/or the URL that can exceed the maximum allowed size, then crash during the limited string unserialization process.

### CODE REVIEW

File: src/qt/pivx/governancemodel.cpp [hunk 1]
```diff
@@ -180,6 +180,14 @@ std::vector<VoteInfo> GovernanceModel::getLocalMNsVotesForProposal(const Proposa
     return localVotes;
 }
 
+OperationResult GovernanceModel::validatePropName(const QString& name) const
+{
+    if (name.toUtf8().size() > PROP_NAME_MAX_SIZE) { // limit
+        return {false, ("Invalid name, maximum size exceeded")};
```
Comment (Fuzzbawls) [inline src/qt/pivx/governancemodel.cpp:186]: missing translation:
```diff
- return {false, ("Invalid name, maximum size exceeded")};
+ return {false, _("Invalid name, maximum size exceeded")};
```

**Review by Fuzzbawls — COMMENTED** [review]
looking good, just missing a string translation.

also, if we want the proposal name field to have the same real-time visual response (red border) in regards to over-length (can happen when using unicode characters), can use this diff:
```diff
Index: src/qt/pivx/createproposaldialog.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/qt/pivx/createproposaldialog.cpp b/src/qt/pivx/createproposaldialog.cpp
--- a/src/qt/pivx/createproposaldialog.cpp	(revision 7342206ca7700d2329b3f6dd06e2cf17b69288f1)
+++ b/src/qt/pivx/createproposaldialog.cpp	(date 1639089820920)
@@ -137,7 +137,7 @@
 
 void CreateProposalDialog::propNameChanged(const QString& newText)
 {
-    bool isValid = !newText.isEmpty() && IsValidUTF8(newText.toStdString());
+    bool isValid = !newText.isEmpty() && IsValidUTF8(newText.toStdString()) && govModel->validatePropName(newText).getRes();
     setCssEditLine(ui->lineEditPropName, isValid, true);
 }
 
```

File: src/qt/pivx/governancemodel.cpp (hunk 1 above)
Comment (furszy) [reply src/qt/pivx/governancemodel.cpp:186]: updated

**Comment by furszy** [comment]
done, quick update. I missed to added it to the text change signal validation.

**Review by Fuzzbawls — APPROVED** [review]
ACK 9d86871ebfdc87e238a61ef46bbbc2ae98393e68

**Review by random-zebra — APPROVED** [review]
utACK 9d86871ebfdc87e238a61ef46bbbc2ae98393e68

**Pull request merged**